org.OpenT2T.Sample.SuperPopular.Thermostat/Nest Thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,179 @@
+    },
+
+    disconnect: function() {
+        console.log('disconnect called.');
+        logDeviceState(this.device);
+    },
+    
+    getTemperatureProperty: function(name, scale, callback) {
+        var options = {
+            protocol: 'https:',
+            host: 'developer-api.nest.com',
+            path: '/devices/thermostats/' + deviceId,
+            headers: {
+                'Authorization': 'Bearer ' + accessToken
+            },
+            method: 'GET'
+        };
+
+        var httpsRequest = https.get(options, function(getRes) {
+            var body = '';
+            getRes.setEncoding('utf8');
+            getRes.on('data', function(data) {
+                body += data;
+            });
+
+            getRes.on('end', function() {
+                if (getRes.statusCode != 200) {
+                    if (callback) {
+                        callback("Error Code:" + getRes.statusCode);
+                        return;
+                    }
+                } else {
+                    var data = JSON.parse(body);
+                    callback(data[name]);
+                    }
+            });
+
+            getRes.on('error', function(e) {
+                callback('error');
+            });
+        });
+
+        httpsRequest.end();
+    },
+
+    setTemperatureProperty: function(name, scale, value, callback) {
+
+        var postData;
+        if (scale == "c"){
+            postData = { name : value };
+        }
+        else {
+            postData = { name : value };
+        }
+        postData = JSON.stringify(postData);
+
+        var options = {
+            protocol: 'https:',
+            host: 'developer-api.nest.com',
+            path: '/devices/thermostats/' + deviceId,
+            headers: {
+                'Authorization': 'Bearer ' + accessToken,
+                'Content-Type': 'application/json',
+                'Accept': 'application/json',
+                'Content-length': Buffer.byteLength(postData),
+            },
+            method: 'PUT',
+        };
+
+        var httpsRequest = https.request(options, (getRes) => {
+            console.log("Request sent");
+            var body = '';
+            getRes.setEncoding('utf8');
+            getRes.on('data', (data) => {
+                body += data;
+            });
+
+            getRes.on('end', () => {
+                if (getRes.statusCode != 200) {
+                    if (callback) {
+                        callback("Error Code:" + getRes.statusCode);
+                        return;
+                    }
+                } else {
+                    var data = JSON.parse(body);
+                    if ((scale == "c") && !!data.target_temperature_c) {
+                        callback(data.target_temperature_c);
+                    }
+                    else if ((scale == "f") && !!data.target_temperature_f) {
+                        callback(data.target_temperature_f);
+                    }
```

Comment:
If the requested unit doesn't match the unit reported by the thermostat, then you need to either do the conversion or report an error. As it is here, the async call never returns in that case.